core-bundle/assets/controllers/jobs-controller.js

```diff
@@ -0,0 +1,67 @@
+import { Controller } from '@hotwired/stimulus';
+import { TurboStreamConnection } from '../modules/turbo-stream-connection';
+
+export default class extends Controller {
+    _turboStreamConnection = new TurboStreamConnection();
+    _runningJobs = false;
+
+    static values = {
+        pendingJobsUrl: String,
+        defaultInterval: Number,
+        maximumInterval: Number,
+        enabled: Boolean,
+    };
+
+    static targets = ['count', 'list'];
+
+    connect() {
+        this._pollInterval = this.defaultIntervalValue;
+        this._timer = null;
+
+        if (this.enabledValue) {
+            this.enable();
+        }
+    }
+
+    enable() {
+        clearTimeout(this._timer);
+        this._poll();
+    }
+
```

Comment:
```suggestion
listTargetDisconnected() {
    clearTimeout(this._timer);
}

```

Once the polling is done and the list-target does not exist anymore (due to the response), we should clear the timeout so it doesn't unnecessarily poll for eternity?

We could also clear this in `_allJobsProcessed()`?

/cc @m-vo

Reply:
Unless this is by design and we want to poll the route once a job has been created, e.g. for new jobs in the future so it'll update the list?

Reply:
This is intentional for exactly this reason. We currently have no stream that adds the target but only a full page reload where jobs are present. The overhead of the few requests for continued polling (with long pauses in between) are negligible I think.